homeassistant/components/renault/services.py

```diff
@@ -0,0 +1,172 @@
+"""Support for Renault services."""
+from __future__ import annotations
+
+from datetime import datetime
+import logging
+from types import MappingProxyType
+from typing import TYPE_CHECKING, Any
+
+import voluptuous as vol
+
+from homeassistant.core import HomeAssistant, ServiceCall
+from homeassistant.helpers import config_validation as cv, device_registry as dr
+
+from .const import DOMAIN
+from .renault_hub import RenaultHub
+from .renault_vehicle import RenaultVehicleProxy
+
+LOGGER = logging.getLogger(__name__)
+
+ATTR_SCHEDULES = "schedules"
+ATTR_TEMPERATURE = "temperature"
+ATTR_VEHICLE = "vehicle"
+ATTR_WHEN = "when"
+
+SERVICE_AC_CANCEL_SCHEMA = vol.Schema(
+    {
+        vol.Required(ATTR_VEHICLE): cv.string,
+    }
+)
+SERVICE_AC_START_SCHEMA = vol.Schema(
+    {
+        vol.Required(ATTR_VEHICLE): cv.string,
+        vol.Required(ATTR_TEMPERATURE): cv.positive_float,
+        vol.Optional(ATTR_WHEN): cv.datetime,
+    }
+)
+SERVICE_CHARGE_SET_SCHEDULE_DAY_SCHEMA = vol.Schema(
+    {
+        vol.Required("startTime"): cv.string,
+        vol.Required("duration"): cv.positive_int,
+    }
+)
+SERVICE_CHARGE_SET_SCHEDULE_SCHEMA = vol.Schema(
+    {
+        vol.Required("id"): cv.positive_int,
+        vol.Optional("activated"): cv.boolean,
+        vol.Optional("monday"): vol.Schema(SERVICE_CHARGE_SET_SCHEDULE_DAY_SCHEMA),
+        vol.Optional("tuesday"): vol.Schema(SERVICE_CHARGE_SET_SCHEDULE_DAY_SCHEMA),
+        vol.Optional("wednesday"): vol.Schema(SERVICE_CHARGE_SET_SCHEDULE_DAY_SCHEMA),
+        vol.Optional("thursday"): vol.Schema(SERVICE_CHARGE_SET_SCHEDULE_DAY_SCHEMA),
+        vol.Optional("friday"): vol.Schema(SERVICE_CHARGE_SET_SCHEDULE_DAY_SCHEMA),
+        vol.Optional("saturday"): vol.Schema(SERVICE_CHARGE_SET_SCHEDULE_DAY_SCHEMA),
+        vol.Optional("sunday"): vol.Schema(SERVICE_CHARGE_SET_SCHEDULE_DAY_SCHEMA),
+    }
+)
+SERVICE_CHARGE_SET_SCHEDULES_SCHEMA = vol.Schema(
+    {
+        vol.Required(ATTR_VEHICLE): cv.string,
+        vol.Required(ATTR_SCHEDULES): vol.All(
+            cv.ensure_list, [SERVICE_CHARGE_SET_SCHEDULE_SCHEMA]
+        ),
+    }
+)
+SERVICE_CHARGE_START_SCHEMA = vol.Schema(
```

Comment:
Consider defining a base schema with the `ATTR_VEHICLE` item and reuse that schema in the other schemas and for all services. Voluptuous schemas can be extended with `Schema.extend`.